### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by tresf**
Mass update of dependencies to latest versions.

@lite1979 @klabarge I'm hoping for some preliminary testing before merging, then some true regression testing before the 2.2.2. release.

Note, this also updates the internal versioning to a more standardized "-SNAPSHOT" version.

### CODE REVIEW

**Comment by tresf** [comment]
Bumping.  Please chime in with some availability for smoke testing this.  I'm only concerned about basic functionality to spot any obvious regressions.

**Comment by lite1979** [comment]
Copying comments from the commit, consolidating into one:

All builds succeeded on the Mac Mini, but Windows and Linux builds failed on the Linux box (AMD 64)

Linux
https://gist.github.com/lite1979/284fde81c40e23e251fa99e7a2df9b58

Windows
https://gist.github.com/lite1979/74b6db4f64063422e18c503eac946b08

**Comment by tresf** [comment]
Quoting comments which were left in the commit:

> ant makeself fails:
> https://gist.github.com/lite1979/284fde81c40e23e251fa99e7a2df9b58

> ant nsis fails:
> https://gist.github.com/lite1979/74b6db4f64063422e18c503eac946b08

Smells like #995, but I'm not sure.  @lite1979 can you send me `java --version` so I can attempt to reproduce?

**Comment by tresf** [comment]
Thanks, patched via 6c969f0.  In https://github.com/qzind/tray/issues/995, we patched what should happen if the version installed is 11.0.16+ and targeting an older version, but this PR targets 11.0.16+, so a system with an older Java installed will suffer an identical bug.  It should be good now. :)

**Comment by lite1979** [comment]
Installation on Windows 10 says it succeeded, but I get a warning after running it, then I can't launch QZ-Tray from either demo.qz.io or from the Start menu.
windowserroroninstall
QZ-TrayLaunchFail

**Comment by lite1979** [comment]
startmenuentry

**Comment by tresf** [comment]
> Installation on Windows 10 says it succeeded, but I get a warning after running it, then I can't launch QZ-Tray from either demo.qz.io or from the Start menu.

Did you update your authenticode cert?  It expired recently.

**Comment by tresf** [comment]
Ok, I think I found the cause: https://github.com/ebourg/jsign/issues/80.  Applying proposed workaround.

**Comment by tresf** [comment]
Hmm... I adapted the workaround to `ant`, but no luck... 

<details>
<summary>Click here for the workaround code that did nothing...</summary>

```diff
diff --git a/ant/windows/installer.xml b/ant/windows/installer.xml
index 375a6fe..c01847c 100644
--- a/ant/windows/installer.xml
+++ b/ant/windows/installer.xml
@@ -126,7 +126,7 @@
     <target name="sign-win-tsa" if="signing.tsaurl" depends="find-jsign">
         <property description="suppress property warning" name="signing.tsaurl" value="something went wrong"/>
         <echo level="info">Signing with timestamp:</echo>
-        <java jar="${jsign.path}" fork="true" failonerror="true">
+        <java classpath="${jsign.path}" classname="net.jsign.JsignCLI" fork="true" failonerror="true">
             <arg value="--name"/>
             <arg value="${project.name}"/>
             <arg value="--url"/>
@@ -150,7 +150,7 @@
     <target name="sign-win-self" unless="signing.tsaurl" depends="find-jsign">
         <echo level="info">Signing without timestamp:</echo>
         <tsa-warning/>
-        <java jar="${jsign.path}" fork="true" failonerror="true">
+        <java classpath="${jsign.path}" classname="net.jsign.JsignCLI" fork="true" failonerror="true">
             <arg value="--name"/>
             <arg value="${project.name}"/>
             <arg value="--url"/>
:...skipping...
diff --git a/ant/windows/installer.xml b/ant/windows/installer.xml
index 375a6fe..c01847c 100644
--- a/ant/windows/installer.xml
+++ b/ant/windows/installer.xml
@@ -126,7 +126,7 @@
     <target name="sign-win-tsa" if="signing.tsaurl" depends="find-jsign">
         <property description="suppress property warning" name="signing.tsaurl" value="something went wrong"/>
         <echo level="info">Signing with timestamp:</echo>
-        <java jar="${jsign.path}" fork="true" failonerror="true">
+        <java classpath="${jsign.path}" classname="net.jsign.JsignCLI" fork="true" failonerror="true">
             <arg value="--name"/>
             <arg value="${project.name}"/>
             <arg value="--url"/>
@@ -150,7 +150,7 @@
     <target name="sign-win-self" unless="signing.tsaurl" depends="find-jsign">
         <echo level="info">Signing without timestamp:</echo>
         <tsa-warning/>
-        <java jar="${jsign.path}" fork="true" failonerror="true">
+        <java classpath="${jsign.path}" classname="net.jsign.JsignCLI" fork="true" failonerror="true">
             <arg value="--name"/>
             <arg value="${project.name}"/>
             <arg value="--url"/>
```
</details>

As it turns out, it's a permissions issue.   Most of the files have very restrictive permissions that only allow an administrator to run them.  This could be a bug with JNA, since we use JNA to setup file permissions on install.  Will have to dive deeper....

Screenshot 2022-08-19 at 8 14 21 PM

**Comment by tresf** [comment]
@lite1979 I opened  #1011 to track this.  From what I can tell, it's actually a bug created from updating our Apache `commons-io` library, which we use to copy the install over to Windows.

What I believe to be happening is that the new version of `FileUtils.copyDirectory(...)` isn't honoring the destination permissions.  I'm not sure how to solve this elegantly.  [Revering to the old `commons-io` lib](https://github.com/qzind/tray/issues/1011#issuecomment-1221204386) is a temporary workaround, but I'm still working on a more permanent fix.

File: src/qz/utils/JsonWriter.java [hunk 1]
```diff
@@ -50,7 +51,7 @@ public static boolean write(String path, String data, boolean overwrite, boolean
             remove(config, append);
         }
 
-        FileUtils.write(f, config.toString(2));
+        FileUtils.write(f, config.toString(2), StandardCharsets.UTF_8);
```
Comment (tresf) [inline src/qz/utils/JsonWriter.java:54]: This change is unrelated, just suppresses a deprecation warning.

**Comment by tresf** [comment]
@lite1979 patched via 8fa41d7.  For those interested, the full discussion is available here: https://stackoverflow.com/questions/73428286.

Since this changes the way we copy the install directory on all platforms, double-checking them after this change is needed.

**Comment by lite1979** [comment]
Installation and basic functionality is now working as expected on MacOS, Windows 10, and Ubuntu 20.04 :)
